flexmeasures/api/v3_0/assets.py

```diff
@@ -326,3 +346,173 @@ def get_chart_data(self, id: int, asset: GenericAsset, **kwargs):
+    # Simplification of checking for create-children access on each of the flexible sensors,
+    # which assumes each of the flexible sensors belongs to the given asset.
+    @permission_required_for_context("create-children", ctx_arg_name="asset")
+    def trigger_schedule(
+        self,
+        asset: GenericAsset,
+        start_of_schedule: datetime,
+        duration: timedelta,
+        belief_time: datetime | None = None,
+        flex_model: dict | None = None,
+        flex_context: dict | None = None,
+        **kwargs,
+    ):
+        """
+        Trigger FlexMeasures to create a schedule for a collection of flexible devices.
+
+        .. :quickref: Schedule; Trigger scheduling job for multiple devices
+
+        Trigger FlexMeasures to create a schedule for this asset.
+        The assumption is that this is a flexible asset containing multiple power sensors.
+
+        In this request, you can describe:
+
+        - the schedule's main features (when does it start, what unit should it report, prior to what time can we assume knowledge)
+        - the flexibility models for the asset's relevant sensors (state and constraint variables, e.g. current state of charge of a battery, or connection capacity)
+        - the flexibility context which the asset operates in (other sensors under the same EMS which are relevant, e.g. prices)
+
+        For details on flexibility model and context, see :ref:`describing_flexibility`.
+        Below, we'll also list some examples.
+
+        .. note:: This endpoint support scheduling an EMS with multiple flexible sensors at once,
+                  but internally, it does so sequentially
+                  (considering already scheduled sensors as inflexible).
+
+        The length of the schedule can be set explicitly through the 'duration' field.
+        Otherwise, it is set by the config setting :ref:`planning_horizon_config`, which defaults to 48 hours.
+        If the flex-model contains targets that lie beyond the planning horizon, the length of the schedule is extended to accommodate them.
+        Finally, the schedule length is limited by :ref:`max_planning_horizon_config`, which defaults to 2520 steps of each sensor's resolution.
+        Targets that exceed the max planning horizon are not accepted.
+
+        The appropriate algorithm is chosen by FlexMeasures (based on asset type).
+        It's also possible to use custom schedulers and custom flexibility models, see :ref:`plugin_customization`.
+
+        If you have ideas for algorithms that should be part of FlexMeasures, let us know: https://flexmeasures.io/get-in-touch/
+
+        **Example request A**
+
+        This message triggers a schedule for a storage asset, starting at 10.00am, at which the state of charge (soc) is 12.1 kWh.
+
+        .. code-block:: json
+
+            {
+                "start": "2015-06-02T10:00:00+00:00",
+                "flex-model": [
+                    {
```

Comment:
This example should be updated with the new schema changes (sensor_flex_model field, etc)

Reply:
That `sensor_flex_model` field is not a field in the public API, but rather a field that gets created when the endpoint validation partly deserializes the `flex-model` field. It essentially belongs to a schema that is internal to our platform, used only for the purpose of *sequential* scheduling. The endpoint logic picks up the still serialized `sensor_flex_model` field and passes it to the job kwargs.